cli/azd/pkg/azapi/deployments.go

```diff
@@ -99,6 +103,18 @@ func NewDeployments(
 	}
 }
 
+func (ds *deployments) CalculateTemplateHash(
+	ctx context.Context,
+	subscriptionId string,
+	template azure.RawArmTemplate) (result armresources.DeploymentsClientCalculateTemplateHashResponse, err error) {
+	deploymentClient, err := ds.createDeploymentsClient(ctx, subscriptionId)
+	if err != nil {
+		return result, fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	return deploymentClient.CalculateTemplateHash(ctx, template, nil)
```

Comment:
The the ARM deployment client already had support for calculating a template hash? Cool!  What algo does it use?

Reply:
the algorithm used by this method is not explicitly mentioned.
https://learn.microsoft.com/en-us/rest/api/resources/deployments/calculate-template-hash?tabs=HTTP